lib/file_items_gateway.rb

```diff
@@ -7,24 +7,29 @@ class FileItemsGateway
 
   def all
     return [] unless File.exist?(FILE_PATH)
-    File.open(FILE_PATH, 'r') do |file|
+    output = File.open(FILE_PATH, 'r') do |file|
+      JSON.parse(file.read, symbolize_names: true)
+    end
+
+    output.map! do |item|
       item_builder = ItemBuilder::Item.new
-      item_builder.items = JSON.parse(file.read, symbolize_names: true)
-      [item_builder.build]
+      item_builder.item_details = item
+      item_builder.build
     end
+    output
   end
 
-  def save(item)
-    item.map! do |part|
+  def save(items)
+    items.map! do |item|
```

Comment:
We've done items.map! more than once across our project (e.g View Summary). Could it be extracted into a module/global method?

Reply:
I think it is better to leave it like that because both use case does different things